include/psa/crypto_types.h

```diff
@@ -380,4 +380,54 @@ typedef uint16_t psa_key_derivation_step_t;
 
 /**@}*/
 
+/** \defgroup pake Password-authenticated key exchange (PAKE)
+ * @{
+ */
+
+/** \brief Encoding of the side of PAKE
+ *
+ * Encodes which side of the protocol is being executed. For more information
+ * see the documentation of individual PSA_PAKE_SIDE_XXX constants.
+ */
+typedef uint16_t psa_pake_side_t;
+
+/** Encoding of input and output indicators for PAKE.
+ *
+ * Some PAKE protocols need to exchange more data than just a single key share.
+ * This type is for encoding additional input and output data for such
+ * protocols.
+ */
+typedef uint8_t psa_pake_data_t;
+
+/** Encoding of the type of the PAKE's primitive.
+ *
+ * Values defined by this standard will never be in the range 0x80-0xff.
+ * Vendors who define additional types must use an encoding in this range.
+ */
+typedef uint8_t psa_pake_primitive_type_t;
+
+/** Encoding of the bitsize for the PAKE's primitive.
+ *
+ * The type and family is not enough to identify the primitive to use in the
+ * PAKE, the implementation needs to know the bitsize too.
+ */
+typedef uint16_t psa_pake_bits_t;
+
+/** Encoding of the PAKE's primitive.
+ *
+ * In most of the PAKEs the primitives are prime order groups, but some of
+ * them might need the ring structure or just are using completely different
+ * algebraic structures (eg. SRP or PQC schemes).
+ */
+typedef uint32_t psa_pake_primitive_t;
+
+/** Encoding of additional options for PAKE.
+ *
+ * This type is for encoding additional options into PAKE cipher suites.
+ * (Options like for example EnvelopeMode in OPAQUE or "Per-User M and N" in
+ * SPAKE2.)
+ */
+typedef uint32_t psa_pake_cipher_suite_options_t;
```

Comment:
Is this intended to be a bitmask?

Reply:
Yes, I thought of this as being a bitmask, but now that you ask, it doesn't have to be. This could potentially be sized down to `uint8_t`.